game/src/main/java/net/demilich/metastone/game/entities/Actor.java

```diff
@@ -193,7 +193,7 @@ public boolean isDestroyed() {
 		if (hasAttribute(Attribute.PERMANENT)) {
 			return hasAttribute(Attribute.DESTROYED);
 		}
-		return getHp() < 1 || super.isDestroyed();
+		return (getHp() < 1 || super.isDestroyed()) && !hasAttribute(Attribute.CANT_BE_DESTROYED);

```

Comment:
This is definitely a problem, and why we have to get rid of that `CANT_BE_DESTROYED` card or rewrite it.